client/cli/src/commands/metadata.rs

```diff
@@ -0,0 +1,61 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Metadata related CLI utilities
+
+use super::MetadataDump;
+use crate::{CliConfiguration, Error, SharedParams};
+use sc_client_api::UsageProvider;
+use sc_service::Arc;
+use sp_api::{Metadata, ProvideRuntimeApi};
+use sp_runtime::traits::Block as BlockT;
+use structopt::StructOpt;
+
+/// Cli subcommand for interacting with the runtime metadata.
+#[derive(Debug, StructOpt)]
+pub enum MetadataSubcommand {
```

Comment:
any other commands you foresee adding here?

Reply:
See this: https://github.com/paritytech/substrate/pull/10450#discussion_r765250616